### ISSUE TITLE
Caddy reverse proxy with slskd

### ISSUE THREAD

**Issue by lmateovf** [slskd/slskd#939]
First of all, thank you for your great program, it works great on my ubuntu server!

May I ask you for a little help with setting up my Caddyfile with slskd. Currently I cannot make it work, I think because of the websockets...

My current caddyfile should look like this:

`
slskd.xxx.org {
    reverse_proxy localhost:5030
}
`
I think, what I need is slskd to be a subdomain, and xxx my main domain. If I'm not mistaken...

Thank you in advance for your time, I hope you can help me...

**Comment by jpdillingham** [comment]
I agree that this is probably the websocket.  Here's a thread I found that might help, specifically the stuff around connection "upgrade".

@tandy-1000 could you share your config?  Pinging because of the discussion from #231 

I can set Caddy up and try it but I won't likely get to it for a while.

**Comment by jpdillingham** [comment]
Maybe something like this? 

```
@websockets {
	header Connection *Upgrade*
	header Upgrade websocket
}
reverse_proxy @websockets localhost:6001
```

**Comment by lmateovf** [comment]
Thank you very much for the fast response.

With this config:
```
#slskd - Soulseek on Ubuntu Server
slskd.xxx.duckndns.org {
@websockets {
	header Connection *Upgrade*
	header Upgrade websocket
}
reverse_proxy @websockets localhost:5030
}
```

Caddy doesn't output any error in the config, but the web only loads a blank page...
I'm sure duckdns allows subdomains, it works with other web apps I have... so it's definetely an error in the Caddyfile... at least in my little knowledge.

If you need I can send you my caddyfile or something... if that helps...

Thanks again

**Comment by jpdillingham** [comment]
Can you `curl` your URL and share the output?  Or alternatively open dev tools in your browser (typically F12) and screenshot the network tab.  

If the problem is only with the socket you should at least get a page with a grey background that complains it couldn't connect.  Getting nothing but a white page indicates that regular HTTP traffic isn't working either.

**Comment by lmateovf** [comment]
Network F12:

![1](https://github.com/slskd/slskd/assets/62722086/a4be1295-bb5e-4055-9c96-2a7211075786)

curl:

![2](https://github.com/slskd/slskd/assets/62722086/f1437f9a-63e6-44bb-b24f-7a1f5b3f9ae4)



It's weird, 'cause on my web browser the certificate it's valid:
![image](https://github.com/slskd/slskd/assets/62722086/3620ecf5-ea87-400d-8ed1-e408408eca3c)

**Comment by Czechball** [comment]
Hello. I used to run slskd behind Caddy, the only directive needed (if I recall correctly) was the one in the first post:
```
domain.com {
    reverse_proxy localhost:5030
}
```
The server in question is not running right now, but I'll see if I can check which config actually worked.

**Comment by lmateovf** [comment]
Hi!
Thank you @Czechball but it doesn't work... That was the first thing I tried...

**Comment by lmateovf** [comment]
Hi, any news?

**Comment by jpdillingham** [comment]
Can you share the contents of the initial `GET /` request? It should look something like:

```html
<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8"/>
        <meta name="viewport" content="width=device-width,initial-scale=1"/>
        <title>slskd</title>
        <script defer="defer" src="/slskd/static/js/main.0da26bea.js"></script>
        <link href="/slskd/static/css/main.0b25c2b6.css" rel="stylesheet">
    </head>
    <body>
        <noscript>You need to enable JavaScript to run this app.</noscript>
        <div id="root"></div>
    </body>
</html>
<script>
    window.urlBase = "/slskd";
    window.port = 5000
</script>

```

**Comment by jpdillingham** [comment]
https://slskd.lmvf.duckdns.org is returning a blank response, so the issue is definitely with duckdns or the caddy configuration; it's not getting far enough for websockets to be a problem.

**Comment by Spilskinanke** [comment]
Was this ever solved? I am seeing the same issue today. Running a local DNS through pihole for a local name for the internal server ip, but it's coming up blank. when curling everything seems A-OK.

**Comment by lmateovf** [comment]
heyyy, hi @Spilskinanke i forgot to mention here that i solved it... this is the correct Caddyfile:

# SLSKD (a soulseek client)
subdomain.domain.duckdns.org {
    reverse_proxy 192.168.1.60:5030 {
        header_up Host {host}
        header_up Upgrade {http_upgrade}
        header_up Connection {http_connection}
    }

**Comment by Spilskinanke** [comment]
Is {host} and {http_upgrade} the actual values or should I substitute that for something else?

**Comment by matthewkdies** [comment]
For anyone curious about this in the future, I added the following to my Caddyfile within a site registration and got it to work:
```
soulseek.mydomain.xyz {
	reverse_proxy soulseek:5030 {
		header_up Upgrade "websocket"
                header_up Connection "Upgrade"
	}
}
```
